xontrib/sh.py

```diff
@@ -29,6 +30,13 @@ def onepath(cmd, **kw):
         first_compatible_shell = None
         check_output_all = ''
         for s in _installed_shells:
+            if len(_installed_shells) == 1:   # don't check syntax for a single shell
```

Comment:
1. When I told about one shell in the list I mean `_shells`. In case of checking `_installed_shells` there could be an issue when the comand will be run in unexpected shell. For example we have shell 1 and 2 in the _shells and on one host the shell 1 exists and on host two the 2 shell is exists (the reasons could be different). The command `! ` will work unexpected differently on every host. We can avoid checking syntax only if `_shell` contains only one shell to create unambiguous behavior.

2. Also this check shouldn't be inside the loop.